### PR TITLE
Work

### PR DESCRIPTION
**Pull request by ritasumtsova**
I've created 4 empty pages Login with fake auth, Departments, Departments and Error.
And added React Router 6 including lazy routes and PrivateRoute component.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,10 +10,12 @@
     "@types/node": "^16.11.31",
     "@types/react": "^18.0.8",
     "@types/react-dom": "^18.0.0",
+    "axios": "^0.27.2",
     "react": "^18.1.0",
     "react-dom": "^18.1.0",
+    "react-router-dom": "^6.3.0",
     "react-scripts": "5.0.1",
-    "typescript": "^4.6.3",
```
Comment (didevgen) [inline package.json:16]: please check ts present in package.json

File: src/App.tsx [hunk 2]
```diff
@@ -1,26 +1,43 @@
...
+        <Route
+          path="/departments/:id"
+          element={
+            (
+              <Suspense fallback={<p>Loading...</p>}>
+                <Department />
+              </Suspense>
+            )
+            }
+        />
```
Comment (didevgen) [inline src/App.tsx]: could you please check against that example?
https://stackoverflow.com/a/70528778

Comment (ritasumtsova) [reply src/App.tsx]: sure, thank you

File: package.json (hunk 1 above)
Comment (ritasumtsova) [reply package.json:16]: my bad

File: src/App.tsx (hunk 2 above)
Comment (ritasumtsova) [reply src/App.tsx]: Checked this one, works as expected. I used this syntax to fix an error

**Uncaught Error: [undefined] is not a <Route> component. All component children of <Routes> must be a <Route> or <React.Fragment>**

It happens when I wrap all my lazy routes in one Suspense component

Comment (didevgen) [reply src/App.tsx]: try wrapping with a fragment as well

Comment (ritasumtsova) [reply src/App.tsx]: I've tried but got the same error.

Also I've seen the same syntax on the React Router [docs](https://stackblitz.com/github/remix-run/react-router/tree/main/examples/lazy-loading?file=src/App.tsx) so I've decided I can write code like that

Comment (didevgen) [reply src/App.tsx]: yeah have seen that too

**Comment by ritasumtsova** [comment]
I've added:
1. TS to the package.json
2. API dir with auth.ts and departments.ts files
3. API calls to the Login, Department, Departments components
4. types dir with the auth, departments, department, employees files

File: src/components/pages/Department/Department.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+  const getDepartmentInfo = async () => {
+    if (id) {
+      await DepartmentsAPI.getDepartmentInfo(id)
+        .then((res) => {
+          setEmployees(res.data.employees);
+        })
+        .catch((err) => console.log(err));
+    }
+  };
```
Comment (didevgen) [inline src/components/pages/Department/Department.tsx]: either move outside of function either memoize

Comment (didevgen) [reply src/components/pages/Department/Department.tsx]: and bad practice to use `await` among with `then`

File: src/components/pages/Departments/Departments.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import React, { useEffect, useState } from 'react';
+import { Link } from 'react-router-dom';
+
+import DepartmentsAPI from '../../../API/Departments';
+import { IDepartments } from '../../../types/departments';
+
+const Departments: React.FC = () => {
+  const [departments, setDepartments] = useState<IDepartments>();
+
+  const getDepartments = async () => {
```
Comment (didevgen) [inline src/components/pages/Departments/Departments.tsx:10]: same here

File: src/components/pages/Departments/Departments.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
+import React, { useEffect, useState } from 'react';
+import { Link } from 'react-router-dom';
+
+import DepartmentsAPI from '../../../API/Departments';
+import { IDepartments } from '../../../types/departments';
+
+const Departments: React.FC = () => {
+  const [departments, setDepartments] = useState<IDepartments>();
```
Comment (didevgen) [inline src/components/pages/Departments/Departments.tsx]: set default value as empty array

File: src/routes/PrivateRoute.tsx [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import { Navigate, Outlet } from 'react-router-dom';
+
+const PrivateRoute: React.FC = () => {
+  return !localStorage.getItem('TOKEN')
```
Comment (didevgen) [inline src/routes/PrivateRoute.tsx]: TOKEN should be in shared constant

**Comment by ritasumtsova** [comment]
Hopefully I followed your point. I fixed the comments, add useMemo hook to the Departments and Department component and changed async func syntax to try catch.

**Comment by ritasumtsova** [comment]
Right, see you. Such a silly mistake.
Should I keep useMemo to better performance or this one is unnecessary here?

**Comment by didevgen** [comment]
of course

**Comment by ritasumtsova** [comment]
Added an empty array as a default value to the useState

File: src/components/pages/Department/Department.tsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    return getDepartmentInfo();
+  }, [id])
+
+  useEffect(() => {}, [departmentInfo]);
+
+  return (
+    <>
+      <h1>Department page</h1>
+      {employees
+        ? employees.map((employee) => {
```
Comment (didevgen) [inline src/components/pages/Department/Department.tsx]: that one is always true now :)

Comment (ritasumtsova) [reply src/components/pages/Department/Department.tsx]: Jeez, thank you :)

Comment (ritasumtsova) [reply src/components/pages/Department/Department.tsx]: fixed this one

**Review by didevgen — APPROVED** [review]

**Comment by ritasumtsova** [comment]
🥳🥳

**Pull request merged**